LoopKit/DosingDecisionStore.swift

```diff
@@ -227,6 +230,33 @@ public struct StoredDosingDecision {
             self.date = date
         }
     }
+
+    public struct DeviceSettings: Codable, Equatable {
+        let name: String
+        let systemName: String
+        let systemVersion: String
+        let model: String
+        let modelIdentifier: String
+        let batteryLevel: Float?
+        let batteryState: BatteryState?
```

Comment:
If `unknown` is a possible value, could this be non-optional?  (put another way, what's the difference between `nil` and `.unknown`?).

Reply:
The two optionals here are in case the device does not support monitoring the battery. I turned this on for Loop, but wanted the option of not specifying it.

Yes, the `batteryState` can be `.unknown`, but there is a difference between `.unknown` from the `UIDevice` and when battery monitoring hasn't been enabled.

Reply:
got it 👍